File: cpp/blind curated 75/implement_strStr.cpp
// Return the index of first occurence of needle in haystack
// -1 if needle not in haystack & 0 if length(needle) = 0

#include <iostream>
#include <string.h>

using namespace std;

int strStr(string haystack, string needle) {
    int h = haystack.length();
    int n = needle.length();

    if (h < n) return -1;

    for (int i=0; i<h; i++) {
        if (haystack.substr(i, n) == needle) {
            return i;
        }
    }

    return -1;
}

int main() {
    cout << strStr("hello", "ll") << endl; //2 
}

File: python/blind curated 75/implement_strStr.py
# Return the index of first occurence of needle in haystack
# -1 if needle not in haystack & 0 if length(needle) = 0


def strStr(haystack: str, needle: str) -> int:
    subLen = len(needle)

    for i in range(0, len(haystack) - subLen + 1):
        if haystack[i : i + subLen] == needle:
            return i

    return -1


print(strStr("hello", "ll"))  # 2
print(strStr("hello", "k"))  # -1
print(strStr("hello", ""))  # 0
